python/caffe/io.py

```diff
@@ -231,11 +231,14 @@ def set_mean(self, in_, mean):
         """
         self.__check_input(in_)
         if mean.ndim == 1:
+            # broadcast pixel
             mean = mean[:, np.newaxis, np.newaxis]
-        mk, mh, mw = mean.shape
-        in_k, in_h, in_w = self.inputs[in_][1:]
-        #if mk != in_k or (mh, mw) != (in_h, in_w) and (mh, mw) != (1, 1):
```

Comment:
@lukeyeager ha this was actually my check for just this but I accidentally commented it out. Your checks cover more cases so when I merge I'll combine them.